edb/server/compiler/dbstate.py

```diff
@@ -277,6 +277,79 @@ def has_ddl(self) -> bool:
         return bool(self.capabilities & enums.Capability.DDL)
 
 
+@dataclasses.dataclass
+class QueryUnitGroup:
+
+    _initialized_single: Optional[bool] = None
+
+    # True if all statements in *sql* can be executed inside a transaction.
+    is_transactional: bool = True
+
+    # All capabilities used by any query units in this group
+    capabilities: enums.Capability = enums.Capability(0)
+
+    # True if it is safe to cache this unit.
+    cacheable: bool = True
+
+    # True if any query unit has tx_id
+    tx_control: bool = False
+
+    # Cardinality of the result set.  Set to NO_RESULT if the
+    # unit represents multiple queries compiled as one script.
+    cardinality: enums.Cardinality = enums.Cardinality.NO_RESULT
+
+    out_type_data: bytes = sertypes.NULL_TYPE_DESC
+    out_type_id: bytes = sertypes.NULL_TYPE_ID.bytes
+    in_type_data: bytes = sertypes.NULL_TYPE_DESC
+    in_type_id: bytes = sertypes.NULL_TYPE_ID.bytes
+    in_type_args: Optional[List[Param]] = None
+    globals: Optional[List[str]] = None
```

Comment:
globals will be fully per QueryUnit, not at all per QueryUnitGroup

Though I guess you can skip worrying about it and I'll fix it when I do argument reworking

Reply:
I see, thanks! I was adding this because global is used in `parse()` of `binary.pyx`, but I’ll leave as it is now.